design-proposals/app-orch-dp-import.md

```diff
@@ -0,0 +1,121 @@
+# Design Proposal: Onboarding Deployment Packages from Github Repositories
+
+Author(s): Scott Baker
+
+Last updated: 2025-04-16
+
+## Abstract
+
+This proposal describes a mechanism for importing a deployment package from an opensource
+application repository into the Orchestrator.
+
+## Proposal
+
+The convention chosen for opensource applications wishing to distribute their deployment packages
+is that deployment packages will be placed in yaml format into well-known `deployment-package/` directories. This
+convention has already been adopted for AI Suite applications at Intel, and endorsed as part of the
+Edge Software Catalog (ESC).
+
+Current mechanisem in 3.0 requires a user to clone the application repository and then import the deployment
+package into the Orchestrator using their web browser. This proposal seeks to eliminate the need
+to clone and import.
+
+This proposal is complicated by the fact that many repositories contain multiple deployment
+packages, and they are often located in subdirectories, and not all AI Suite repositories use the
+same subdirectory naming convention.
+
+There are two variants of this proposal:
+
+### Sub-Proposal - Import a specific deployment package from a repository (requires more GUI work)
+
+- The existing `Import Deployment Package` screen shall be updated with an `Import from URL` button.
```

Comment:
```suggestion
- The existing `Import Deployment Package <https://docs.openedgeplatform.intel.com/edge-manage-docs/main/user_guide/package_software/import_deployment.html>` screen shall be updated with an `Import from URL` button.
```

Reply:
Should we call it 'Import from URL' or 'Import from Github Repository', as we expect that the URL points to a GitHub repo?

Reply:
updated.